unity-client/Assets/Scripts/MainScripts/DCL/Components/UI/UIScreenSpace/UIScreenSpace.cs

```diff
@@ -185,6 +179,7 @@ void InitializeCanvas()
             if (!scene.isPersistent)
             {
                 UpdateCanvasVisibility();
+                CommonScriptableObjects.allUIHidden.OnChange += AllUIHidden_OnChange;
```

Comment:
Had to move this line here because `scene` is still not populated in the constructor now.

Reply:
I'm concerned about having multiple subscriptions to this event by mistake. Maybe we can use the `initialized` flag or unsubscribe preemptively

Reply:
`InitializeCanvas()` is only called if the already existing `initialized` flag is false. So there's no chance of multiple subscriptions in this case. If that was the case, we could be creating many canvases by mistake too, so I can safely bet on this.